pyomo/solver/ipopt.py

```diff
@@ -0,0 +1,517 @@
+
+
+class ipoptSolutionLoader(SolutionLoaderBase):
+    pass
+
+
+ipopt_command_line_options = {
+    'acceptable_compl_inf_tol',
+    'acceptable_constr_viol_tol',
+    'acceptable_dual_inf_tol',
+    'acceptable_tol',
+    'alpha_for_y',
+    'bound_frac',
+    'bound_mult_init_val',
+    'bound_push',
+    'bound_relax_factor',
+    'compl_inf_tol',
+    'constr_mult_init_max',
+    'constr_viol_tol',
+    'diverging_iterates_tol',
+    'dual_inf_tol',
+    'expect_infeasible_problem',
+    'file_print_level',
+    'halt_on_ampl_error',
+    'hessian_approximation',
+    'honor_original_bounds',
+    'linear_scaling_on_demand',
+    'linear_solver',
+    'linear_system_scaling',
+    'ma27_pivtol',
+    'ma27_pivtolmax',
+    'ma57_pivot_order',
+    'ma57_pivtol',
+    'ma57_pivtolmax',
+    'max_cpu_time',
+    'max_iter',
+    'max_refinement_steps',
+    'max_soc',
+    'maxit',
+    'min_refinement_steps',
+    'mu_init',
+    'mu_max',
+    'mu_oracle',
+    'mu_strategy',
+    'nlp_scaling_max_gradient',
+    'nlp_scaling_method',
+    'obj_scaling_factor',
+    'option_file_name',
+    'outlev',
+    'output_file',
+    'pardiso_matching_strategy',
+    'print_level',
+    'print_options_documentation',
+    'print_user_options',
+    'required_infeasibility_reduction',
+    'slack_bound_frac',
+    'slack_bound_push',
+    'tol',
+    'wantsol',
+    'warm_start_bound_push',
+    'warm_start_init_point',
+    'warm_start_mult_bound_push',
+    'watchdog_shortened_iter_trigger',
+}
+
+
+@SolverFactory.register('ipopt_v2', doc='The ipopt NLP solver (new interface)')
+class ipopt(SolverBase):
+    CONFIG = ipoptConfig()
+
+    def __init__(self, **kwds):
+        self._config = self.CONFIG(kwds)
+        self._writer = NLWriter()
+        self._writer.config.skip_trivial_constraints = True
+        self._solver_options = self._config.solver_options
+
+    def available(self):
+        if self.config.executable.path() is None:
+            return self.Availability.NotFound
+        return self.Availability.FullLicense
+
+    def version(self):
+        results = subprocess.run(
+            [str(self.config.executable), '--version'],
+            timeout=1,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True,
+        )
+        version = results.stdout.splitlines()[0]
+        version = version.split(' ')[1].strip()
+        version = tuple(int(i) for i in version.split('.'))
+        return version
+
+    @property
+    def writer(self):
+        return self._writer
+
+    @property
+    def config(self):
+        return self._config
+
+    @config.setter
+    def config(self, val):
+        self._config = val
+
+    @property
+    def solver_options(self):
+        return self._solver_options
+
+    @solver_options.setter
+    def solver_options(self, val: Dict):
+        self._solver_options = val
+
+    @property
+    def symbol_map(self):
+        return self._symbol_map
```

Comment:
Ideally we want to delete this. The new interface should not need this. If the legacy interface needs it, create and declare.